runtime/build.gradle

```diff
@@ -108,3 +107,11 @@ if (project.findProperty("publication_repository") == "space") {
 tasks.withType(KotlinNativeCompile).matching { it.name.toLowerCase().contains("test") }.configureEach {
     it.dependsOn(tasks.withType(Sign))
 }
+
+tasks.withType(KotlinNativeCompile).configureEach {
+    compilerOptions.freeCompilerArgs.addAll(
+            "-opt-in=kotlin.experimental.ExperimentalNativeApi",
+            "-opt-in=kotlin.native.runtime.NativeRuntimeApi",
+            "-opt-in=kotlinx.cinterop.ExperimentalForeignApi",
```

Comment:
Maybe you could continue using `languageSettings.optIn` DSL but move it to `nativeMain` source set?

Reply:
Unfortunately, this approach doesn't work. The opt-ins are not passed to the dependent source sets automatically.
This causes:
```
Inconsistent settings for Kotlin source sets: 'androidNativeArm32Main' depends on 'nativeMain'
  'androidNativeArm32Main': set of opt-in annotations in use is []
  'nativeMain': set of opt-in annotations in use is [kotlin.experimental.ExperimentalNativeApi, kotlin.native.runtime.NativeRuntimeApi, kotlinx.cinterop.ExperimentalForeignApi]
  The dependent source set must use all opt-in annotations that its dependency uses.
```